src/app/shared/header/header.component.html

```diff
@@ -3,9 +3,16 @@
     <div class="icon-section">
       <div class="title">Mobite</div>
       <img src="/assets/app-logo.svg"/></div>
-    <div>
-      <app-button class="sign-up" [buttonStyle]="signUpButtonStyle" (click)="openSignupPage.emit()">Sign Up</app-button>
-      <app-button class="login" (click)="openLoginPage.emit()">Login</app-button>
+    <div *ngIf="!signedInUser else UserSignedIn">
+      <app-button class="sign-up" [buttonStyle]="signUpButtonStyle" (click)="openSignupPage()">Sign Up</app-button>
+      <app-button class="login" (click)="openLoginPage()">Login</app-button>
     </div>
+    <ng-template #UserSignedIn>
+      <div class="loggedIn">
+        You are signed in!
```

Comment:
Could we display something like "Welcome, UserName" here?

Reply:
For that I need backend support to return me the username.

Reply:
Username and email should be inside the jwt payload, perhaps you could use something like this https://www.npmjs.com/package/jwt-decode ?

Reply:
Good idea. Got it working.